pkg/kubernetes/apis/loadtest/v1/types.go

```diff
@@ -42,14 +44,16 @@ type LoadTestPodsStatus struct {
 
 // LoadTestSpec is the spec for a LoadTest resource
 type LoadTestSpec struct {
-	Type            LoadTestType `json:"type"`
-	Overwrite       bool         `json:"overwrite"`
-	MasterConfig    ImageDetails `json:"masterConfig"`
-	WorkerConfig    ImageDetails `json:"workerConfig"`
-	DistributedPods *int32       `json:"distributedPods"`
-	TestFile        string       `json:"testFile"`
-	TestData        string       `json:"testData,omitempty"`
-	EnvVars         string       `json:"envVars,omitempty"`
+	Type            LoadTestType  `json:"type"`
+	Overwrite       bool          `json:"overwrite"`
+	MasterConfig    ImageDetails  `json:"masterConfig"`
+	WorkerConfig    ImageDetails  `json:"workerConfig"`
+	DistributedPods *int32        `json:"distributedPods"`
+	TestFile        string        `json:"testFile"`
+	TestData        string        `json:"testData,omitempty"`
+	EnvVars         string        `json:"envVars,omitempty"`
+	TargetURL       string        `json:"targetURL,omitempty"`
```

Comment:
Are target url and duration allowed to be empty? As I see locust test doesn't do anything without these params specified. 
I think we shouldn't create a loadtest spec for locust type if those params are empty.

Reply:
They both can be empty yes.
For targetURL, you can use full URLs inside the test script.
For duration, Locust allows you to run tests forever if you want to.